ncm-hostsfile/src/main/perl/hostsfile.pm

```diff
@@ -2,31 +2,137 @@
+
+Returns error in case of a failure.
+
+=back
+
+=head1 RESOURCES
+
+=over 4
+
+=item * C<< /system/network/domainname >>
+
+When specifying hosts within the entries nlist, if a hostname is not FQDN
+and there are no aliases defined, then an alias will be automatically
+created using an FQDN formed by joining the shortname with
+this domain.
+
+=item * C<< /software/components/hostsfile/file >>
+
+The filename to modify, defaults to C<< /etc/hosts >>.
+
+=item * C<< /software/components/hostsfile/entries >>
+
+An nlist, keyed by hostname. The value of each hostname is an nlist
+containing the following structure:
+
+=over 4
+
+=item C<ipaddr>
+
+The IP address of the host.
+
+=item I<aliases>
+
+A string value of aliases. Multiple aliases should be whitespace separated.
+
+=item C<comment>
+
+A comment to append to the line within /etc/hosts.
+
+=back
+
+=item * C<< /software/components/hostsfile/takeover >>
+
+A boolean. If false (the default), then pre-existing host lines in the file
+which are not tagged with the "NCM" comment will be preserved.
+If takeover is true,
+then pre-existing entries for hosts will be taken over and declared to be
+under NCM control.
 
-#
-# a few standard statements, mandatory for all components
-#
+=back
+
+=head1 EXAMPLES
+
+Example - configuration defined like this:
+
+  include 'software/components/hostsfile';
+  "/software/components/hostsfile/active" = true;
+  "/software/components/hostsfile/file" = "/etc/hosts.local";
+  "/software/components/hostsfile/entries" = nlist (
+    "tsmstor601.cern.ch", nlist(
+        "ipaddr", "192.168.1.101",
+        "comment", "TSM DB disks"),
+    "tsmstor602.cern.ch", nlist(
+        "ipaddr", "192.168.1.102",
+        "comment", "TSM Staging disks"),
+    );
+
+will modify the /etc/hosts.local file from:
+
+  # Do not remove the following line, or various programs
+  # that require network functionality will fail.
+  127.0.0.1               localhost.localdomain localhost
+  137.138.45.75           lxfsec1604.cern.ch
+
+to:
+
+  # Generated by Quattor component hostsfile
+  # Do not remove the following line, or various programs
+  # that require network functionality will fail.
+  127.0.0.1               localhost.localdomain localhost
+  137.138.45.75           lxfsec1604.cern.ch
+  192.168.1.101   tsmstor601.cern.ch tsmstor601 # NCM TSM DB disks
+  192.168.1.101   tsmstor602.cern.ch tsmstor602 # NCM TSM Staging disks
+
+The syntax below is also possible:
+
+  "/software/components/hostsfile/entries/tsmstor603" = nlist(
+      "ipaddr", "192.168.1.103",
+      "comment", "TSM more disks"
+  );
+
+or:
+
+  "/software/components/hostsfile/entries/tsmstor603/ipaddr" = "192.168.1.103";
+  "/software/components/hostsfile/entries/tsmstor603/comment" = "Testing";
+
+=cut
+
+package NCM::Component::hostsfile;
 
 use strict;
 use base 'NCM::Component';
 
 our $EC = LC::Exception::Context->new->will_store_all;
-our $NoActionSupported = 1;
+our $NoActionSupported = 0;
```

Comment:
AFAIK LC works fine with NoAction (CAF just calls through to them) so I don't think it's necessary to disable this unless we know there's a piece of code that is not safe?

Reply:
My mistake!